### PR TITLE
MEP: Wiring for Customize toolbar schema

### PR DESCRIPTION
**Pull request by sharvaniharan**
**Phab:** https://phabricator.wikimedia.org/T297818

### CODE REVIEW

File: app/src/main/java/org/wikipedia/analytics/eventplatform/CustomizeToolbarEvent.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+@Serializable
+@SerialName("/analytics/mobile_apps/android_customize_toolbar_interaction/1.0.0")
+class CustomizeToolbarEvent(private var is_anon: Boolean,
+                            private var is_rfm_enabled: Boolean,
+                            private var source: String,
+                            private var favorites_order: List<Int>,
+                            private var menu_order: List<Int>,
+                            private var time_spent_ms: Int) : TimedEvent(STREAM_NAME) {
+
+    fun logCustomization(source: String, favorites_order: List<Int>, menu_order: List<Int>, is_rfm_enabled: Boolean) {
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/analytics/eventplatform/CustomizeToolbarEvent.kt:17]: Would it be possible to make the parameters have a naming style like this: `favoritesOrder`?

**Review by cooltey — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/wikipedia/analytics/eventplatform/CustomizeToolbarEvent.kt (hunk 1 above)
Comment (sharvaniharan) [reply app/src/main/java/org/wikipedia/analytics/eventplatform/CustomizeToolbarEvent.kt:17]: The parameter names have to match the MEP tables Cooltey ... and they have a snake_case convention..

File: app/src/main/java/org/wikipedia/analytics/eventplatform/TimedEvent.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+open class TimedEvent(@Transient private val streamName: String = "") : Event(streamName) {
+
+    @Transient
+    private var startTime = 0L
+
+    @Transient
+    private var pauseTime = 0L
+
+    @Transient
+    val duration = System.currentTimeMillis() - startTime
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/analytics/eventplatform/TimedEvent.kt:16]: @sharvaniharan  this field is never updated, and will therefore always be incorrect. You probably intended for this to be a `get()` property?

Comment (sharvaniharan) [reply app/src/main/java/org/wikipedia/analytics/eventplatform/TimedEvent.kt:16]: Thank you.. fixed here: #3237

**Pull request merged**